crypto/evp_extra/p_pqdsa_asn1.c

```diff
@@ -18,6 +18,40 @@ static void pqdsa_free(EVP_PKEY *pkey) {
   pkey->pkey.pqdsa_key = NULL;
 }
 
+static int pqdsa_get_priv_raw_seed(PQDSA_KEY *key, const PQDSA *pqdsa,
+                                  uint8_t *out,size_t *out_len) {
+  if (key->seed == NULL) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_NO_KEY_SET);
+    return 0;
+  }
+
+  if (*out_len != key->pqdsa->keygen_seed_len) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_BUFFER_TOO_SMALL);
+    return 0;
+  }
+
+  OPENSSL_memcpy(out, key->seed, pqdsa->keygen_seed_len);
+  *out_len = pqdsa->keygen_seed_len;
+  return 1;
+}
+
+static int pqdsa_get_priv_raw_key(PQDSA_KEY *key, const PQDSA *pqdsa,
+                                  uint8_t *out,size_t *out_len) {
+  if (key->private_key == NULL) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_NOT_A_PRIVATE_KEY);
+    return 0;
+  }
+
+  if (*out_len < key->pqdsa->private_key_len) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_BUFFER_TOO_SMALL);
+    return 0;
+  }
+
+  OPENSSL_memcpy(out, key->private_key, pqdsa->private_key_len);
+  *out_len = pqdsa->private_key_len;
+  return 1;
+}
+
 static int pqdsa_get_priv_raw(const EVP_PKEY *pkey, uint8_t *out,
                                    size_t *out_len) {
   if (pkey->pkey.pqdsa_key == NULL) {
```

Comment:
We should also check that `pkey` and `out_len` are not NULL here as well.

Reply:
Thank you, added in [c307952](https://github.com/aws/aws-lc/pull/2194/commits/c30795274cf00bed09803b8f5fe6114112fb1422)